packages/base-manager/src/manager-base.ts

```diff
@@ -664,91 +663,83 @@ export abstract class ManagerBase implements IWidgetManager {
+      if (model.buffers) {
+        const bufferPaths = model.buffers.map((b: any) => b.path);
+        // put_buffers expects buffers to be DataViews
+        const buffers = model.buffers.map(
+          (b: any) => new DataView(decode[b.encoding](b.data))
+        );
+        put_buffers(model.state, bufferPaths, buffers);
+      }
 
-          // If the model has already been created, set its state and then
-          // return it.
-          if (this.has_model(model_id)) {
-            return this.get_model(model_id)!.then((model) => {
-              // deserialize state
-              return (model.constructor as typeof WidgetModel)
-                ._deserialize_state(modelState || {}, this)
-                .then((attributes) => {
-                  model.set_state(attributes); // case 2
-                  return model;
-                });
-            });
-          }
+      // If the model has already been created, set its state and then
+      // return it.
+      if (this.has_model(model_id)) {
+        const model = await this.get_model(model_id);
+        // deserialize state
+        const attributes = await (
+          model.constructor as typeof WidgetModel
+        )._deserialize_state(modelState || {}, this);
+        model.set_state(attributes); // case 2
+        return model;
+      }
 
-          const modelCreate: IModelOptions = {
-            model_id: model_id,
-            model_name: model.model_name,
-            model_module: model.model_module,
-            model_module_version: model.model_module_version,
-          };
-          if (Object.prototype.hasOwnProperty.call(live_comms, 'model_id')) {
-            // live comm
-            // This connects to an existing comm if it exists, and
-            // should *not* send a comm open message.
-            return this._create_comm(this.comm_target_name, model_id).then(
-              (comm) => {
-                modelCreate.comm = comm;
-                return this.new_model(modelCreate); // No state, so safe wrt. case 1
-              }
-            );
-          } else {
-            return this.new_model(modelCreate, modelState); // case 1
-          }
-        })
-      );
+      const modelCreate: IModelOptions = {
+        model_id: model_id,
+        model_name: model.model_name,
+        model_module: model.model_module,
+        model_module_version: model.model_module_version,
+      };
+      if (Object.prototype.hasOwnProperty.call(live_comms, 'model_id')) {
+        // live comm
+        // This connects to an existing comm if it exists, and
+        // should *not* send a comm open message.
+        const comm = await this._create_comm(this.comm_target_name, model_id);
+        modelCreate.comm = comm;
+        return this.new_model(modelCreate); // No state, so safe wrt. case 1
+      } else {
+        return this.new_model(modelCreate, modelState); // case 1
+      }
     });
 
-    return all_models;
+    return Promise.all(all_models);
```

Comment:
I think we can get rid of this variable and just return as the previous code was doing it